### PR TITLE
[CARBONDATA-746] Add spark-sql CLI support for spark2.1 carbon integration

### PR DESCRIPTION
**Pull request by jackylk**
Added classes for using CarbonSession in spark-sql CLI
Usage: set the SPARK_HOME environment variable and execute bin/carbon-spark-sql

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1007/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1008/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1011/

File: integration/spark2/src/main/scala/org/apache/spark/sql/hive/cli/CarbonSQLCLIDriver.scala [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      val storePath = System.getenv("CARBON_HOME") + "/bin/carbonsqlclistore"
+      val warehouse = System.getenv("CARBON_HOME") + "/warehouse"
+      val carbon = SparkSession
+          .builder()
+          .master(System.getProperty("spark.master"))
+          .appName("CarbonSQLCLIDriver")
+          .config("spark.sql.warehouse.dir", warehouse)
+          .getOrCreateCarbonSession(storePath, storePath)
+
+      hiveContext = new CarbonContext(carbon)
```
Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/spark/sql/hive/cli/CarbonSQLCLIDriver.scala]: I guess it is not required to create context here, just use `carbon.sqlContext` is enough.

File: integration/spark2/src/main/scala/org/apache/spark/sql/hive/cli/CarbonContext.scala [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hive.cli
+
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.apache.spark.sql.hive.HiveContext
+
+class CarbonContext(
```
Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/spark/sql/hive/cli/CarbonContext.scala]: I guess it is not required to create context , just use the context available in  CarbonSession would be enough.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1031/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**